### PR TITLE
Fix binary version linking

### PR DESCRIPTION
**Pull request by alexhung**
Replace 'go build' with 'GoReleaser build' so the 'ldflags' is set in only one place at .goreleaser.yml

Tested by:
- run `make install` and then `TF_LOG=INFO terraform plan` and checks for log with `Provider version: x.y.z`
- also verify the call home request using Charles

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -1,7 +1,10 @@
 TEST?=./...
-TARGET_ARCH=$(shell go env GOOS)_$(shell go env GOARCH)
+GO_ARCH=$(shell go env GOARCH)
+TARGET_ARCH=$(shell go env GOOS)_${GO_ARCH}
+ifeq ($(GO_ARCH), amd64)
+GORELEASER_ARCH=${TARGET_ARCH}_$(shell go env GOAMD64)
```
Comment (alexhung) [inline GNUmakefile:5]: GoReleaser adds `_v1` from `GOAMD64` when architecture is `amd64`.

File: GNUmakefile [hunk 2]
```diff
@@ -48,7 +43,7 @@ attach:
 
 acceptance: fmtcheck
 	export TF_ACC=true && \
-		go test -ldflags="-X '${PKG_VERSION_PATH}.Version=${NEXT_VERSION}-test'" -v -p 1 -parallel 20 -timeout 20m ./pkg/...
+		go test -v -p 1 -parallel 20 -timeout 20m ./pkg/...
```
Comment (chb0github) [inline GNUmakefile:46]: I really at this point would like to print out the version when the TF provider runs.

**Review by chb0github — CHANGES_REQUESTED** [review]
output version number please

File: GNUmakefile (hunk 2 above)
Comment (alexhung) [reply GNUmakefile:46]: It does already but I need to revert this so the correct version is logged.

**Pull request merged**